### PR TITLE
Add the ability to use regexp for tags in docker images

### PR DESCRIPTION
**Pull request by IizunaK**
In Leboncoin we started using Concourse for our CD but we needed to make some adjustment for some resources, so here is the change we had to do for the docker-image-resource.

For tools like kubernetes which can be setup not to force pull if some image with the same name is already present,
it can be useful to just use a regexp on a tag and select the latest tag matching the regexp.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@IizunaK Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=238)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=238) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@IizunaK Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=238)!

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +7,7 @@ COPY assets/ /assets
 RUN go build -o /assets/check github.com/concourse/docker-image-resource/cmd/check
 RUN go build -o /assets/print-metadata github.com/concourse/docker-image-resource/cmd/print-metadata
 RUN go build -o /assets/ecr-login github.com/concourse/docker-image-resource/vendor/github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cmd
+RUN apk add --update gcc g++
```
[the remark this answers is not in the document]
Comment (IizunaK) [reply Dockerfile]: If I don't add it I can't run the go tests.
You can try it yourself, but as far as I tried, these dependencies are missing.

**Comment by IizunaK** [comment]
So what do you think of this PR ?

**Comment by IizunaK** [comment]
Hello ?

**Comment by vito** [comment]
@IizunaK Sorry, PRs slowed down while we were redoing our build process. I'm getting back to them now and can take a look at this today.

**Comment by Typositoire** [comment]
Dunno if this is dead but I'm also down for having this. Right now I have a task that outputs labels after the GET, this would simplify my life :P

**Comment by IizunaK** [comment]
For me it is not dead, I don't know on Pivotal side though, they seem
pretty busy

Le mar. 18 déc. 2018 à 15:31, Yann David <notifications@github.com> a
écrit :

**Comment by vito** [comment]
Hey, really sorry for the silence on this. Honestly I started reviewing and then saw more changes than I expected and got distracted before submitting. But now that I look at it again, I don't think I want to change the behavior of this resource substantially anymore because we're planning to phase it out in favor of the [`registry-image` resource](https://github.com/concourse/registry-image-resource). This resource is already very complicated and we're making a conscious effort to prevent it from getting worse.

I think the upcoming [spaces feature](https://github.com/concourse/concourse/projects/2) will also have a big impact on this and *might* make it unnecessary, because tags will now be represented as spaces that jobs can run against.

I'm going to close this and recommend that you run on a fork instead for now - sorry again for the delay.

**Pull request closed without merging**